tools/lib/helpers.py

```diff
@@ -14,7 +15,7 @@ class RE:
 
   INDEX = r'-?[0-9]+'
   SLICE = r'(?P<start>{})?:?(?P<end>{})?:?(?P<step>{})?'.format(INDEX, INDEX, INDEX)
-  SEGMENT_RANGE = r'{}(?:--|/)?(?P<slice>({}))?/?(?P<selector>([qras]))?'.format(ROUTE_NAME, SLICE)
+  SEGMENT_RANGE = re.compile(r'{}(?:(--|/)(?P<slice>({})))?(?:/(?P<selector>([qras])))?'.format(ROUTE_NAME, SLICE))
```

Comment:
AFAIK, there's no way to include a non-capturing group with a named group and have the group be an empty string instead of None if it doesn't exist